### PR TITLE
[examples] Add price sim

### PR DESCRIPTION
**Pull request by Chepelau**

### CODE REVIEW

**Comment by MarekHauzr** [comment]
@Chepelau please rename the "sim" dir to something more explanatory... thanks :)

**Comment by MarekHauzr** [comment]
@Chepelau can you make a summary at the end of the notebook that will say something
```
for strike 1000 and current price 800, CALL pool size 20_000ETH
the premium shifts by ...% when we trade option of size 50ETH (0.25% of the pool)
the premium shifts by ...% when we trade option of size 100ETH (0.5% of the pool)
the premium shifts by ...% when we trade option of size 200ETH (1% of the pool)
the premium shifts by ...% when we trade option of size 500ETH (2.5% of the pool)

for strike 1000 and current price 800, PUT pool size 20_000_000 USD
the premium shifts by ...% when we trade option of size 50ETH (0.25% of the pool)
the premium shifts by ...% when we trade option of size 100ETH (0.5% of the pool)
the premium shifts by ...% when we trade option of size 200ETH (1% of the pool)
the premium shifts by ...% when we trade option of size 500ETH (2.5% of the pool)

*for higher detail (buy vs sell, different impacts in maturities, etc) adjust the parameters in the notebook
```

**Comment by Chepelau** [comment]
Just pushed some changes, let's talk about this tommorrow

File: examples/price_impact_simulation/aux.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+MATURITY = 1672527600  # 31/12/2022 -> matures in one year
+STRIKE_PRICE = 2305843009213693952000  # 1000 * 2**61
+CURRENT_PRICE = 1_200  # of ETH
+CURRENT_PRICE_2 = 800
+
+
+def pretty_dict(_dict, indent=4):
+    if not isinstance(_dict, dict):
+        raise TypeError('Not a dict')
+    print(json.dumps(_dict, indent=indent))
```
Comment (MarekHauzr) [inline examples/price_impact_simulation/aux.py:26]: why not use [pretty_print](https://docs.python.org/3/library/pprint.html)

File: examples/price_impact_simulation/aux.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    print('\n')
+
+
+async def show_current_premium(amm, trade_size, return_res=False):
+    if not isinstance(amm, StarknetContract):
+        raise TypeError('Not a Starknet Contract')
+
+    # Functions are only called, not invoked, so the state wont be changes but it'll return current premium
+    prem_CALL = await amm.trade(account_id=USER_ID, option_type=0, strike_price=STRIKE_PRICE, maturity=MATURITY, side=0, option_size=trade_size).call()
+    prem_PUT = await amm.trade(account_id=USER_ID, option_type=1, strike_price=STRIKE_PRICE, maturity=MATURITY, side=1, option_size=trade_size).call()
```
Comment (MarekHauzr) [inline examples/price_impact_simulation/aux.py:107]: why is side=0 for call and side=1 for put?

**Pull request merged**